src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt

```diff
@@ -0,0 +1,108 @@
+package chip.setuppayload
+
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/** Parser for scanned QR code or manual entry code.  */
+class SetupPayloadParser {
+  /**
+   * Returns [SetupPayload] parsed from the QR code string. If an invalid element is included
+   * in the QRCode Parse result, SetupPayloadException occurs. Refer to [SetupPayload] for the
+   * description of the invalid element.
+   */
+  @Throws(UnrecognizedQrCodeException::class, SetupPayloadException::class)
+  fun parseQrCode(qrCodeString: String): SetupPayload {
+    return fetchPayloadFromQrCode(qrCodeString, false)
+  }
+
+  /**
+   * Returns [SetupPayload] parsed from the QR code string.
+   *
+   * @param qrCodeString the QRCode for commissioning device.
+   * @param allowInvalidPayload Allow invalid payload components. If this value is true, payload
+   * element validation is not checked. Consider saying that the payload must still parse
+   * correctly, but this skips validation of the content past parsing (i.e. it does not validate
+   * ranges for individual elements). Refer to [SetupPayload] for the description of the
+   * invalid element.
+   */
+  @Throws(UnrecognizedQrCodeException::class, SetupPayloadException::class)
+  fun parseQrCode(qrCodeString: String, allowInvalidPayload: Boolean): SetupPayload {
+    return fetchPayloadFromQrCode(qrCodeString, allowInvalidPayload)
+  }
+
+  /**
+   * Returns [SetupPayload] parsed from the manual entry code string. If an SetupPINCode has
+   * invalid value, SetupPayloadException occurs. Refer to [SetupPayload] for the description
+   * of the invalid element.
+   */
+  @Throws(InvalidEntryCodeFormatException::class, SetupPayloadException::class)
+  fun parseManualEntryCode(entryCodeString: String): SetupPayload {
+    return fetchPayloadFromManualEntryCode(entryCodeString, false)
```

Comment:
"fetch" as a verb implies going somewhere. This is just a conversion/parsing.

Curious why it is called "fetchXxxx"